src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -164,6 +167,17 @@ public string StorageConnectionString
         /// </summary>
         public TimeSpan? FunctionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether function invocations will timeout when
+        /// a <see cref="FunctionTimeout"/> is specified and a debugger is attached. False by default.
+        /// </summary>
+        public bool TimeoutWhileDebugging { get; set; }
+
+        /// <summary>
+        /// When true, an exception is thrown when a function timeout expires.
+        /// </summary>
+        public bool ThrowExceptionOnFunctionTimeout { get; set; }
```

Comment:
When using this with the cooperative model as well (CancellationToken) how long after we signal cancellation will we throw? I.e., in this model what is the graceful shutdown period?